tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -1143,17 +1175,20 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
 
     def lookupField(name: String): Option[CheckedField] =
       fields.get(name).orElse(superClass.flatMap(_.lookupField(name)))
+
+    def lookupStaticField(name: String): Option[CheckedField] =
+      staticFields.get(name)
   }
 
   private object CheckedClass {
     private def checkedField(fieldDef: FieldDef) = {
-      val FieldDef(false, Ident(name, _), tpe, mutable) = fieldDef
-      new CheckedField(name, tpe, mutable)
+      val FieldDef(static, Ident(name, _), tpe, mutable) = fieldDef
+      new CheckedField(static, name, tpe, mutable)
     }
   }
 
-  private class CheckedField(val name: String, val tpe: Type,
-      val mutable: Boolean)
+  private class CheckedField(val static: Boolean, val name: String,
+      val tpe: Type, val mutable: Boolean)
```

Comment:
You should check for static field collisions (I should probably have said this when you introduced static fields). See #2627.